sw/device/lib/dif/dif_pattgen.c

```diff
@@ -0,0 +1,141 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_pattgen.h"
+
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/dif/dif_base.h"
+
+#include "pattgen_regs.h"  // Generated.
+
+dif_result_t dif_pattgen_configure_channel(
+    const dif_pattgen_t *pattgen, dif_pattgen_channel_t channel,
+    dif_pattgen_channel_config_t config) {
+  if (pattgen == NULL || config.polarity > kDifPattgenPolarityHigh ||
```

Comment:
optional nit: The check on config.polarity probably doesn't have the best readability. A common pattern for enums is to define a final "count" value (like `kDifPattgenPolarityCount`) and check against that, if this relational operator is desired, that is.

But... not a big deal. At least the polarity enum seems unlikely to expand.

Reply:
Fixed.